code/shared-mobility/shared-mobility-smrt.js

```diff
@@ -158,3 +158,210 @@ $(document).on("knack-scene-render.scene_339", function () {
+      })
+        .then(function (res) {
+          // Remove spinner after invoice item record is created
+          $("#" + id + "-spinner").remove();
+          // Clear all checkboxes
+          $(".table-checkboxes").each(function (event) {
+            $(this).prop("checked", false);
+          });
+          // refetch view
+          Knack.views[viewKey].model.fetch();
+        })
+        .fail(function () {
+          $("#" + id + "-spinner").remove();
+          console.error("Failed to add invoice to impoundments")
+        });
+    });
+}
+
+function addCheckboxes(view) {
+  // Add the checkbox to to the header to select/unselect all
+  $("#" + view.key + ".kn-table thead tr").prepend(
+    '<th class="table-checkboxes-parent"><input class="table-checkboxes" type="checkbox"></th>'
+  );
+
+  $("#" + view.key + ".kn-table thead input").change(function () {
+    $("." + view.key + ".kn-table tbody tr input").each(function () {
+      $(this).attr(
+        "checked",
+        $("#" + view.key + ".kn-table thead input").attr("checked") !==
+          undefined
+      );
+    });
+  });
+
+  // Add a checkbox to each row in the table body
+  $("#" + view.key + ".kn-table tbody tr").each(function () {
+    $(this).prepend(
+      '<td class="table-checkboxes-parent"><input class="table-checkboxes" type="checkbox"></td>'
+    );
+  });
+
+  function toggleCheckbox($checkbox) {
+    $checkbox.is(":checked")
+      ? $checkbox.prop("checked", false)
+      : $checkbox.prop("checked", true);
+  }
+
+  // Add click event handler to checkbox parent to check/uncheck child box
+  $("#" + view.key + " .table-checkboxes-parent").click(function (event) {
+    // If table header, check/uncheck all checkboxes
+    if ($(this).is("th")) {
+      var $headerCheckbox = $($(this).children()[0]);
+      toggleCheckbox($headerCheckbox);
+      $(
+        "#" + view.key + " td.table-checkboxes-parent input.table-checkboxes"
+      ).each(function () {
+        $headerCheckbox.is(":checked")
+          ? $(this).prop("checked", true)
+          : $(this).prop("checked", false);
+      });
+    } else {
+      var $checkbox = $($(this).children()[0]);
+      toggleCheckbox($checkbox);
+    }
+  });
+
+  // Restore default checkbox toggle
+  $("#" + view.key + " .table-checkboxes").click(function (event) {
+    var $checkbox = $(this);
+    toggleCheckbox($checkbox);
+  });
+
+}
+
+function appendSubmitButton(buttonString, selector, handler, viewKey) {
+  var id = buttonString.toLowerCase().split(" ").join("-");
+  $(selector).append(
+    '<a id="' +
+      id +
+      '" class="kn-button"><span class="icon is-small"><i class="fa fa-check"></i></span><span>' +
+      buttonString +
+      "</span></a>"
+  );
+
+  $("#" + id).click(function () {
+    handler(id, viewKey);
+  });
+}
+
+
+// the view with the Provider Impoundments table
+$(document).on("knack-view-render.view_1044", function(event, view, data) {
+  addCheckboxes(view);
+
+  appendSubmitButton(
```

Comment:
This is probably an artifact from the finance & purchasing code but I noticed this submit button will continuously spin if no checkboxes are selected and you click it. This is probably not the most important thing but I wanted to mention in case it affects the other app too.

Reply:
Plus one to Mike's observation about the Infinite spinner

https://github.com/user-attachments/assets/3ce7dfb3-50b5-4ca4-bde9-0dd729b121ed